pywemo/discovery.py

```diff
@@ -91,3 +95,54 @@ def device_from_uuid_and_location(uuid, mac, location,
                           rediscovery_enabled=rediscovery_enabled)
 
     return None
+
+
+def hostname_lookup(hostname):
+    """Resolve a hostname into an IP address."""
+    try:
+        # The {host} must be resolved to an IP address; if this fails, this will
+        # throw a socket.gaierror.
+        host_address = gethostbyname(hostname)
+
+        # Reset {host} to the resolved address.
+        LOG.debug(
+            'Resolved hostname %s to IP address %s.',
+            hostname, host_address)
+        return host_address
+
+    except gaierror:
+        # The {host}-as-hostname did not resolve to an IP address.
+        LOG.debug(
+            'Could not resolve hostname %s to an IP address.',
+            hostname)
+        return None
+
+
+def setup_url_for_address(host, port):
+    """Determine setup.xml url for a given host and port pair."""
+
+    # Force hostnames into IP addresses
+    try:
+        # Attempt to register {host} as an IP address; if this fails ({host} is not an IP address),
+        # this will throw a ValueError.
+        ip_address(host)
+        is_hostname = False
+    except ValueError:
+        # The provided {host} should be treated as a hostname.
+        is_hostname = True
+
+    if is_hostname:
+        host_address = hostname_lookup(host)
+        if host_address is not None:
+            host = host_address
+        else:
+            return None
```

Comment:
I'm slightly concerned about a use case where a call is made with a host that we don't understand - but would work perfectly well with the underlying library.

So I think if we don't think what they passed is an ip address, but the lookup didn't work - I think the best fallback is just to try what the gave us - rather then returning none.

It might well not work - but it seems better to try doing what they asked - rather than deciding they were wrong ourselves!

This is the logic that was in my suggested code sample.

Reply:
I think that's fair. I can't think of any situation where this would matter - it's either an IP, or a resolvable hostname, or an invalid hostname which shouldn't be queried - but I'm OK with that.